eZ/Publish/Core/Repository/BookmarkService.php

```diff
@@ -97,16 +101,22 @@ public function loadBookmarks(int $offset = 0, int $limit = 25): BookmarkList
     {
         $currentUserId = $this->getCurrentUserId();
 
-        $list = new BookmarkList();
-        $list->totalCount = $this->bookmarkHandler->countUserBookmarks($currentUserId);
-        if ($list->totalCount > 0) {
-            $bookmarks = $this->bookmarkHandler->loadUserBookmarks($currentUserId, $offset, $limit);
-
-            $list->items = array_map(function (Bookmark $bookmark) {
-                return $this->repository->getLocationService()->loadLocation($bookmark->locationId);
-            }, $bookmarks);
+        $filter = new Filter();
+        try {
+            $filter
+                ->withCriterion(new Criterion\IsBookmarked($currentUserId))
+                ->withSortClause(new SortClause\BookmarkId(Query::SORT_DESC))
+                ->sliceBy($limit, $offset);
+
+            $result = $this->repository->getlocationService()->find($filter, []);
+        } catch (BadStateException $e) {
+            return new BookmarkList();
         }
```

Comment:
This exception should probably be logged. We had plenty of situations in the past where people asked why something isn't working as it should, while we "swallowed" the exception.

Reply:
Added logging as suggested. Please have a new look @Steveb-p